src/components/screens/vacation-requests/vacation-requests-table/vacation-requests-table-toolbar/toolbar-form/form-fields.tsx

```diff
@@ -0,0 +1,120 @@
+import {
+  Button,
+  FormControl,
+  FormLabel,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+  TextField
+} from "@mui/material";
+import getLocalizedVacationType from "../../../../../../utils/vacation-type-utils";
+import { VacationType } from "../../../../../../generated/client";
+import { ChangeEvent, Dispatch } from "react";
+import DateRangePicker from "../../../../../generics/date-range-picker";
+import { VacationData } from "../../../../../../types";
+import { SetStateAction } from "jotai";
+import { DateTime } from "luxon";
+/**
+ * Component properties
+ */
+interface FormFieldsProps {
+  vacationData: VacationData;
+  setVacationData: Dispatch<SetStateAction<VacationData>>;
+  dateTimeNow: DateTime;
+  initialStartDate: DateTime | undefined;
+  initialEndDate: DateTime | undefined;
+  readyToSubmit: boolean;
+}
+
+/**
+ * FormFields
+ *
+ * @param props FormFieldsProps
+ */
+const FormFields = (props: FormFieldsProps) => {
+  const {
+    vacationData,
+    setVacationData,
+    dateTimeNow,
+    initialEndDate,
+    initialStartDate,
+    readyToSubmit
+  } = props;
+
+  /**
+   * Set dates to vacationData
+   *
+   * @param startDate
+   * @param endDate
+   * @param days
+   */
+  const setDates = (
+    startDate: DateTime | undefined,
+    endDate: DateTime | undefined,
+    days: number
+  ) => {
+    setVacationData({
+      ...vacationData,
+      startDate: startDate,
+      endDate: endDate,
+      days: days
+    });
+  };
+
+  return (
+    <FormControl sx={{ width: "100%" }}>
+      <FormLabel>Vacation Type</FormLabel>
```

Comment:
Localizations are missing from this branch, but could be created into another task.

Reply:
Yeah, thought about off-loading localization work to Sami, if he wants to do it.